databricks-kube/src/context.rs

```diff
@@ -32,15 +32,28 @@ pub struct OperatorConfiguration {
 
 impl Context {
     pub fn get_databricks_url_token(&self) -> Option<(String, String)> {
-        let latest_secret = Self::latest_api_secret(self.api_secret_store.clone())?;
+        let latest_secret = Self::latest_store(self.api_secret_store.clone())?;
 
         let url = latest_secret.get("databricks_url")?;
         let token = latest_secret.get("access_token")?;
         Some((url.to_string(), token.to_string()))
     }
 
-    fn latest_api_secret(api_secret_store: Arc<Store<Secret>>) -> Option<BTreeMap<String, String>> {
-        api_secret_store
+    pub fn get_timeout_params(&self) -> Option<(String, String, String)> {
+        let latest_secret = Self::latest_store(self.api_secret_store.clone())?;
+
+        let default_poll_interval = &String::from("30");
+        let default_timeout_seconds = &String::from("15");
+        let default_requeue_retry = &String::from("300");
+
+        let poll_interval_millis = latest_secret.get("poll_interval_millis").unwrap_or(default_poll_interval);
```

Comment:
I don't know how exactly to achieve this (the fun part!), but it would be awesome you could reify the secret contents as `OperatorConfiguration`. Some tips:

- Instead of deriving Default, try implementing it manually with the defaults from above. [std Default docs](https://doc.rust-lang.org/std/default/trait.Default.html)
- So -- I don't know the easiest way to go from `Secret` to `OperatorConfiguration`. You can:
  - Make one after these lines with all the params
  - Marshal the `data` map into a JSON string then reify it as `OperatorConfiguration` 
  - Here are k8s-openapi docs for [Secret](https://docs.rs/k8s-openapi/latest/k8s_openapi/api/core/v1/struct.Secret.html)

Reply:
So one thing I thought of which I pushed up just to show is that I'm taking the contents of the secret and pasting it into the map that lives inside the `OperatorConfiguration` when we call `get_databricks_url_token` (sort of like the 2nd route you mentioned about marshalling the `data` map).

It gets us a "sort-of" `Secret` to `OperatorConfiguration` conversion on demand, though it feels a bit forced in my eyes

Reply:
I see what you mean. Hang tight, writing up some suggestions 😃